### ISSUE TITLE
/torch/install/bin/lua: bad argument #2 to '?' (out of range at /torch/pkg/torch/generic/Tensor.c:913)

### ISSUE THREAD

**Issue by netxiao** [OpenNMT/OpenNMT#156]
I used configuration file to train from a saved model,  when the training is to end an epoch，I got errors as:

> [03/03/17 23:24:11 INFO] Epoch 6 ; Iteration 108050/108070 ; Learning rate 0.5000 ; Source tokens/s 867 ; Perplexity 12.99	
> /torch/install/bin/lua: bad argument #2 to '?' (out of range at /torch/pkg/torch/generic/Tensor.c:913)
> stack traceback:
> 	[C]: in ?
> 	[C]: in function '__index'
> 	.../work/torch/install/share/lua/5.2/onmt/train/Trainer.lua:125: in function 'trainEpoch'
> 	.../work/torch/install/share/lua/5.2/onmt/train/Trainer.lua:272: in function 'train'
> 	train.lua:118: in function 'main'
> 	train.lua:123: in main chunk
> 	[C]: in function 'dofile'
> 	...work/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
> 	[C]: in ?

my config file contents:
```
start_epoch = 6
train_from =model_common1_epoch5.t7
#continue = true

max_batch_size = 80
optim = sgd
learning_rate = 0.5


end_epoch = 13
start_decay_at = 6

layers = 4
brnn = true
rnn_size = 1024
#word_vec_size = 350 

gpuid = 1
curriculum = 1

save_every = 5000

pre_word_vecs_enc = data_common/train/vec_src-embeddings-200.t7
pre_word_vecs_dec = data_common/train/vec_tgt-embeddings-200.t7
```
I have upgraded last version yesterday, but still have this problem.

**Comment by guillaumekln** [comment]
Did you retrain on different data?

**Comment by netxiao** [comment]
yes，train two different data。

**Comment by guillaumekln** [comment]
Are you sure you did not use `-continue`? Otherwise, did you set the `-start_iteration` option even though the dataset is not the same?

**Comment by netxiao** [comment]
I used the `-continue` parameter in epoch1-2, in epoch4  used once `-start_iteration`,maybe I used the wrong dataset to cause this problem.

**Comment by guillaumekln** [comment]
If you retrain on another dataset from an **intermediate** checkpoint, I don't think it makes sense to use `-continue` nor `-start_iteration`.

**Comment by netxiao** [comment]
thanks for your help!
